autopush/endpoint.py

```diff
@@ -873,11 +902,15 @@ def _success(self, result):
     #                    Utility Methods
     #############################################################
     def _validate_auth(self, uaid):
-        """Validates the Authorization header in a request"""
+        """Validates the Authorization header in a request
+
+        Validate the given request using HAWK.
+        """
         secret = self.ap_settings.crypto_key
-        hashed = self.request.headers.get("Authorization", "").strip()
-        key = generate_hash(secret, uaid)
-        return validate_hash(key, self.request.body, hashed)
+
+        return hawkauthlib.check_signature(self.request,
+                                           secret,
```

Comment:
Should this be `generate_hash(secret, uaid)`? It looks like that's the shared secret, not the `crypto_key` itself.